### PR TITLE
external/stdk: Update stdk to v1.0.10

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
update stdk source code and modify porting code for it.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [55302781f0f25e5a0c339f01391c927d0201f5ef] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [55302781f0f25e5a0c339f01391c927d0201f5ef] - Code Rule Check OK.

File: external/stdk/patch/0001-patch-for-1.0.9_fix-overwrite-issue-return-value-and.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+diff --git a/src/easysetup/iot_easysetup_d2d.c b/src/easysetup/iot_easysetup_d2d.c
+index 2d55f60..79ce1b3 100644
+--- a/src/easysetup/iot_easysetup_d2d.c
++++ b/src/easysetup/iot_easysetup_d2d.c
+@@ -444,7 +444,7 @@ static iot_error_t _es_keyinfo_handler(struct iot_context *ctx, char *in_payload
+ 		goto exit;
+ 	}
+ 
+-	for (i = 0, j = 0; i < sizeof(rand_asc); i += 2, j++) {
++	for (i = 0, j = 0; i < sizeof(rand_asc) - 1; i += 2, j++) {
```
Comment (xixidodo) [inline external/stdk/patch/0001-patch-for-1.0.9_fix-overwrite-issue-return-value-and.patch]: let's sync up with 1.0.10,     then this patch is not needed anymore

Comment (SRCN-LiQin) [reply external/stdk/patch/0001-patch-for-1.0.9_fix-overwrite-issue-return-value-and.patch]: Good idea, working on it.

**Comment by sunghan-chang** [comment]
stdk and stdk example make build errors. Could you check them?

**Comment by sunghan-chang** [comment]
When I build TizenRT with artik053/hello, I meet belows
```
make[3]: Entering directory '/root/tizenrt/apps/examples/stdk_smart_lamp'
cp: cannot stat '/root/tizenrt/os/../external/stdk/st-device-sdk-c/src/include/profile/json/device_info_example.json': No such file or directory
xxd: /device_info.json: No such file or directory
cp: cannot stat '/root/tizenrt/os/../external/stdk/st-device-sdk-c/src/include/profile/json/onboarding_config_example.json': No such file or directory
xxd: /onboarding_config.json: No such file or directory
```
with distclean
```
make[3]: Entering directory '/root/tizenrt/apps/examples/stdk_smart_lamp'
cp: cannot stat '/root/tizenrt/os/../external/stdk/st-device-sdk-c/src/include/profile/json/device_info_example.json': No such file or directory
xxd: /device_info.json: No such file or directory
cp: cannot stat '/root/tizenrt/os/../external/stdk/st-device-sdk-c/src/include/profile/json/onboarding_config_example.json': No such file or directory
xxd: /onboarding_config.json: No such file or directory
make[3]: Leaving directory '/root/tizenrt/apps/examples/
...
make[2]: Entering directory '/root/tizenrt/external/stdk'
/bin/sh: 1: cannot create ./st-device-sdk-c/src/iot_root_ca.c: Directory nonexistent
/bin/sh: 1: cannot create ./st-device-sdk-c/src/iot_root_ca.c: Directory nonexistent
sed: can't read ./st-device-sdk-c/src/iot_root_ca.c: No such file or directory
sed: can't read ./st-device-sdk-c/src/iot_root_ca.c: No such file or directory
rm: cannot remove './st-device-sdk-c/src/iot_root_ca.c.bak': No such file or directory
```

**Comment by seinfra** [comment]
Target : [9c7e7d0711ae5771ddd8bb9dc4634f49e33d4d4f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c7e7d0711ae5771ddd8bb9dc4634f49e33d4d4f] - Code Rule Check OK.

**Comment by SRCN-LiQin** [comment]
version 1.0.10 updated, with new patch for less modification in iot-core. 
and the  build error issue on non-stdk case will be handled with new PR.

**Pull request merged**